Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -0,0 +1,259 @@
+import macro from 'vtk.js/Sources/macro';
+
+const MAX_POINTS = 2;
+
+export default function widgetBehavior(publicAPI, model) {
+  model.classHierarchy.push('vtkDistanceWidgetProp');
+  let isDragging = null;
+  const direction = [0, 0, 0];
+
+  // --------------------------------------------------------------------------
+  // Display 2D
+  // --------------------------------------------------------------------------
+
+  publicAPI.setDisplayCallback = (callback) =>
+    model.representations[0].setDisplayCallback(callback);
+
+  // --------------------------------------------------------------------------
+  // Interactor events
+  // --------------------------------------------------------------------------
+
+  function ignoreKey(e) {
+    return e.altKey || e.controlKey || e.shiftKey;
+  }
+
+  function updateConeDirection(behavior, callData) {
+    let bv = behavior;
+    if (bv === 3) {
+      const test = publicAPI.computeWorldToDisplay(
+        model.renderer,
+        model.widgetState.getHandle1List()[0].getOrigin()[0],
+        model.widgetState.getHandle1List()[0].getOrigin()[1],
+        model.widgetState.getHandle1List()[0].getOrigin()[2]
+      );
+      const mousePos = publicAPI.computeDisplayToWorld(
+        model.renderer,
+        callData.position.x,
+        callData.position.y,
+        test[2]
+      );
+      for (let i = 0; i < 3; i++) {
+        direction[i] =
+          model.widgetState.getHandle1List()[0].getOrigin()[i] - mousePos[i];
+      }
+      bv = 0;
+    } else {
+      const modifier = bv === 1 ? 1 : -1;
+      bv -= 1;
+      for (let i = 0; i < 3; i++)
+        direction[i] =
+          (model.widgetState.getHandle1List()[0].getOrigin()[i] -
```

Comment:
Maybe you can use an explicit variable to avoid calling `getHandle1List()[0].getOrigin()` at each loop.